src/Microsoft.OData.Core/UriParser/Aggregation/ApplyBinder.cs

```diff
@@ -116,6 +129,7 @@ private IEdmTypeReference CreateAggregateExpressionTypeReference(SingleValueNode
                                     expressionPrimitiveKind));
                     }
 
+                case AggregationMethod.VirtualPropertyCount:

```

Comment:
Your observation is correct. I added this line to be able to use CreateAggregateExpressionTypeReference for both cases, but ended up not using it in the $count one. I'll correct it.